crates/data_components/src/dynamodb/provider.rs

```diff
@@ -99,6 +98,33 @@ impl DynamoDBTableProvider {
             .table_name(table_name.to_string())
             .send()
             .await
+            .map_err(|err| {
+                // Convert the SdkError to a more user-friendly error
+                let source = match err.into_source() {
```

Comment:
```suggestion
                let source = err.into_source().map_err(Into::into)?;
```
nit: This is cleaner (if it works?).

Reply:
Yeah - this will be a great simplification for the syntax, unfortunately this it can't be used like this here(